### ISSUE TITLE
g is not a function

### ISSUE THREAD

**Issue by windhamg** [netsage-project/netsage-sankey-panel#12]
I'm trying to use this plugin with Grafana v8.4.10.

When the panel attempts to render the graph, I get an error: `g is not a function. (In 'g(u.nodeColor)', 'g' is undefined)` 

<img width="1617" alt="Screen Shot 2022-08-05 at 5 16 57 PM" src="https://user-images.githubusercontent.com/2934459/183225754-7c68ae83-5af0-48c5-b310-99a10c8b08a8.png">

Any ideas as to what might be wrong? Thanks!

**Comment by AlxLifanov** [comment]
Similar problem, but 'g is not a function is only one aspect.
Grafana 8.5.3, mysql data source, plugin 1.0.6.

Input data is:
![image](https://user-images.githubusercontent.com/65549908/196161099-74ab896a-62df-4255-ad3e-dfbfab0b63a3.png)

If i leave WHERE as time-limiting macro - nothing works.

The only way i can make it draw something is the following:

![image](https://user-images.githubusercontent.com/65549908/196161579-a7c18dfd-5b80-47dc-9cd8-4f09e14048d7.png)

![image](https://user-images.githubusercontent.com/65549908/196161629-d292b5f1-c68e-423f-8f88-1523ed9a6e79.png)

You can see that "time" is really used as "from" and "value" is both as "To" and "Metric"
From and To are ignored.
I suppose, embedded Grafana field names are used incorrectly.

**Comment by luusluus** [comment]
Try Format as Table

**Comment by uladzimir-yushkevich** [comment]
I see same issue:
![image](https://user-images.githubusercontent.com/79853336/233697283-310f1542-a372-408d-b84d-32fe1e2f1397.png)
[v9.4.7 (4add91f03d)](https://github.com/grafana/grafana/blob/main/CHANGELOG.md)
version 1.0.6

**Comment by pjrwlazlo** [comment]
I have the same issue ("g is not a function" error when I try to add data). Has there been any updates on this?

**Comment by KatrinaTurner** [comment]
Hey everyone, sorry for the late reply, I didn't get a notification and am just now seeing these.
A couple things, 
1) This plugin was not intended to be used for Time Series data, so that could be why it's having an issue
2) The thickness of the lines is from the first number field it finds, which I realize could be causing issues.  I can put in a ticket to allow the option to choose the value field.

In terms of the "g is not a function" error, that is unfortunately a super generic Grafana error and I can't really debug based on that.  If you could screenshot your console, that would make it easier for me to debug.
Also, what data sources is everyone using?  I developed it using primarily an Elastic Data Source, so I don't always catch bugs from the query outputs of other data sources.
Mahalo!

**Comment by gdiana** [comment]
When I got G is not a function I accepted the json then it gave P is not a function

**Comment by im-architeq** [comment]
Hello,
unfortunately I get the same error for my Grafana 10.0.0 .. Could you please fix it? Thanks in advance!

<img width="222" alt="PisNotAFunction" src="https://github.com/netsage-project/netsage-sankey-panel/assets/63162803/54a30d14-b290-4fe0-9c76-c7ecf0e9348b">

**Comment by ffisc** [comment]
Same issue here. This can be reproduced with a 5-row dataset: 
[g_not_function.csv](https://github.com/user-attachments/files/17072950/g_not_function.csv)

**Comment by gdiana** [comment]
Hi, Try this, I got it to work. Must select Table and format data correctly. Note tect data must be reformeted as number.

https://github.com/netsage-project/netsage-sankey-panel/issues/18#issuecomment-1681681573

**Comment by gdiana** [comment]
Ok I imported your data table and this is what I got

![image](https://github.com/user-attachments/assets/0211ad88-f7db-4828-b4b7-61685f278097)

I used Transform Data 1- Convert field Type to convert count as text to number

![image](https://github.com/user-attachments/assets/f81be199-fe6f-497c-b8bf-4444638afc63)

All worked ok

**Comment by ffisc** [comment]
Thank you and sorry for the late reply - still no luck. The data looks fine when viewing as table, but the plugin somehow can't handle it.

<img width="1215" alt="image" src="https://github.com/user-attachments/assets/b16810f6-391c-4686-adbd-bd6c0dbffaf5">

<img width="1223" alt="image" src="https://github.com/user-attachments/assets/a5002573-ad38-4e4f-853d-d72e8488052f">

**Comment by gdiana** [comment]
Hi Felipe,

What version are you using. In my post above you will see an option called **VALUE FIELD** which I set to **COUNT.**

But I do not see this option in your screenshot? It appears between The Node padding and Layout Iterations sliders but in yours it is missing? See below. Maybe it is somewhere else but you need to find it and set as I have done.

![image](https://github.com/user-attachments/assets/66ab421e-877f-408c-9d87-120cdba41d64)

**Comment by ffisc** [comment]
Indeed, I had also noted the lack of the `Value Field` selection, but had thought it might be related to the lack of data.

I was using `1.0.6`. I've updated to `1.1.3` and now it works! 🎉

**Comment by gdiana** [comment]
Great, alls well that ends well ☺️
